src/opentimelineio/serializableCollection.h

```diff
@@ -129,6 +130,19 @@ SerializableCollection::children_if(
                     out.push_back(valid_child);
                 }
             }
+            else if (auto timeline = dynamic_cast<Timeline*>(child.value))
+            {
+                const auto valid_children =
+                    timeline->children_if<T>(error_status, search_range);
```

Comment:
Thanks for putting this together, @darbyjohnston!  This is pretty straightforward.  @meshula I think we'll need to return to this argument because it isn't specified what space the `search_range` is in.  In the case of a serializable collection, which doesn't have a temporal relationship between its children, I could see dropping this argument entirely.  For timelines, if its in the global space, then the offset should be subtracted first to get into the space of the `.tracks` stack.

I also sort of wonder if Serializable collection needs children_if/clip_if?  or needs one that does this kind of recursion?  I wonder if it makes more sense to drop it or make it always shallow search so that you can then operate on the results of this in a consistent way.  Any thoughts or opinions?  @jminor @reinecke 

We could leave it as is - or as it is after this PR or I could see an argument for dropping it.

Reply:
1) I assume that in the case of a timeline, the range would be in the timeline's space, so wouldn't need a subtraction?

2) I do wonder how deep down the rabbit hole we should go in addressing the initial report. I'm inclined  to suggest solving for Timeline is sufficient for 0.15.0, but more fundamental thinking about *_if seems prime territory for exploration in 0.16.0.